addons/controllers/calico/calicoconfig_controller.go

```diff
@@ -46,3 +88,181 @@ func (r *CalicoConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&cniv1alpha1.CalicoConfig{}).
 		Complete(r)
 }
+
+// ReconcileCalicoConfig reconciles CalicoConfig data values secrets
+func (r *CalicoConfigReconciler) ReconcileCalicoConfig(
+	ctx context.Context,
+	cluster *clusterapiv1beta1.Cluster,
+	calicoConfig *cniv1alpha1.CalicoConfig) (_ ctrl.Result, retErr error) {
+
+	var patchCRD bool
+
+	patchHelper, err := clusterapipatchutil.NewHelper(calicoConfig, r.Client)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	// patch CalicoConfig before returning the function
+	defer func() {
+		// patchCRD will be true if finalizer or owner reference is added/removed
+		if patchCRD {
+			r.Log.Info("Patching CalicoConfig")
+			if err := patchHelper.Patch(ctx, calicoConfig.DeepCopy()); err != nil {
```

Comment:
question: From my understanding, we do not need deep copy here. Are there any specific reason of having it here ?

Reply:
Removed from here, thanks.